src/game_map.cpp

```diff
@@ -847,6 +834,28 @@ int Game_Map::GetTerrainTag(int x, int y) {
 		return 1;
 	}
 
+	// Terrain tag wraps on looping maps
+	if (Game_Map::LoopHorizontal()) {
+		x = RoundX(x);
+	}
+	if (Game_Map::LoopVertical()) {
+		y = RoundY(y);
+	}
+
+	// RPG_RT always uses the terrain of the first lower tile
+	// for out of bounds coordinates.
+	unsigned chip_index = 0;
+
+	if (Game_Map::IsValid(x, y)) {
+		const auto chip_id = map->lower_layer[x + y * GetWidth()];
+		chip_index = ChipIdToIndex(chip_id);
+
+		// Apply tile substitution
+		if (chip_index >= BLOCK_E_INDEX && chip_index < NUM_LOWER_TILES) {
+			chip_index = map_info.lower_tiles[chip_index - BLOCK_E_INDEX] + 18;
```

Comment:
is this 18 something different to BLOCK_E_INDEX?

Reply:
Nope, just an oversight. Fixed Thanks!